bot/resources/tags/async-await.md

```diff
@@ -2,18 +2,18 @@
 
 Python provides the ability to run multiple tasks and coroutines simultaneously with the use of the `asyncio` library, which is included in the Python standard library.
 
-This works by running these coroutines in an event loop, where the context of which coroutine is being run is switches periodically to allow all of them to run, giving the appearance of running at the same time. This is different to using threads or processes in that all code is run in the main process and thread, although it is possible to run coroutines in threads.
+This works by running these coroutines in an event loop, where the context of the running coroutine switches periodically to allow all the coroutines to run, thus giving the appearance of running at the same time. This is different to using threads or processes in that all code runs in the main process and thread, although it is possible to run coroutines in threads.
 
 To call an async function we can either `await` it, or run it in an event loop which we get from `asyncio`.
 
-To create a coroutine that can be used with asyncio we need to define a function using the async keyword:
+To create a coroutine that can be used with asyncio we need to define a function using the `async` keyword:
 ```py
 async def main():
     await something_awaitable()
 ```
-Which means we can call `await something_awaitable()` directly from within the function. If this were a non-async function this would have raised an exception like: `SyntaxError: 'await' outside async function`
+Which means we can call `await something_awaitable()` directly from within the function. If this were a non-async function, this would have raised the exception `SyntaxError: 'await' outside async function`
 
-To run the top level async function from outside of the event loop we can get an event loop from `asyncio`, and then use that loop to run the function:
+To run the top level async function from outside the event loop, we can get the event loop from `asyncio` and then use it to run the function:
```

Comment:
Can we feature creep and take the time to recommend `asyncio.run()` instead?

Reply:
I'm not at all familiar with `asyncio` so I'd need more info as to what to write.

Reply:
You can see the [documentation for some more context](https://docs.python.org/3/library/asyncio-task.html#asyncio.run), but I'd recommend something like the following:
> To run the top level async function from outside of the event loop we need to use `asyncio.run()` like this:
> ```py
> import asyncio
> 
> async def main():
>     await something_awaitable()
> 
> asyncio.run(main())

Then in the note after that change the reference of `run_until_complete()` to `asyncio.run()`.

Reply:
Will look into this tonight :+1:

Reply:
Updated as part of [33b4f42](https://github.com/python-discord/bot/pull/1836/commits/33b4f42b481e6b5e1686f66803dc916595e4b457)